File: src/main/java/tech/zolhungaj/amqapi/servercommands/objects/RankedLeaderboardEntry.java
package tech.zolhungaj.amqapi.servercommands.objects;

import org.jetbrains.annotations.NotNull;

public record RankedLeaderboardEntry(
        Integer score,
        String name,
        Integer position
) implements Comparable<RankedLeaderboardEntry>
{
    @Override
    public int compareTo(@NotNull RankedLeaderboardEntry o) {
        return this.position() - o.position();
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/objects/RankedLeaderboardEntry.kt
package tech.zolhungaj.amqapi.servercommands.objects

import com.squareup.moshi.Json

@JvmRecord
data class RankedLeaderboardEntry(
    val score: Int,
    @Json(name = "name") val playerName: String?,
    val position: Int
) : Comparable<RankedLeaderboardEntry> {
    override fun compareTo(other: RankedLeaderboardEntry): Int {
        return this.position - other.position
    }
}
